api/include/opentelemetry/trace/propagation/http_trace_context.h

```diff
@@ -0,0 +1,276 @@
+    int tmp2 = HexToInt(trace_flags[1]);
+    if (tmp1 < 0 || tmp2 < 0)
+      return TraceFlags(0);  // check for invalid char
+    uint8_t buf = tmp1 * 16 + tmp2;
+    return TraceFlags(buf);
+  }
+
+private:
+  // Converts a single character to a corresponding integer (e.g. '1' to 1), return -1
+  // if the character is not a valid number in hex.
+  static uint8_t HexToInt(char c)
+  {
+    if (c >= '0' && c <= '9')
+    {
+      return (int)(c - '0');
+    }
+    else if (c >= 'a' && c <= 'f')
+    {
+      return (int)(c - 'a' + 10);
+    }
+    else if (c >= 'A' && c <= 'F')
+    {
+      return (int)(c - 'A' + 10);
+    }
+    else
+    {
+      return -1;
+    }
+  }
+
+  static void InjectTraceParent(const SpanContext &span_context, T &carrier, Setter setter)
+  {
+    char trace_id[32];
+    TraceId(span_context.trace_id()).ToLowerBase16(trace_id);
+    char span_id[16];
+    SpanId(span_context.span_id()).ToLowerBase16(span_id);
+    char trace_flags[2];
+    TraceFlags(span_context.trace_flags()).ToLowerBase16(trace_flags);
+    // Note: This is only temporary replacement for appendable string
+    std::string hex_string = "00-";
+    for (int i = 0; i < 32; i++)
+    {
+      hex_string += trace_id[i];
+    }
+    hex_string += "-";
+    for (int i = 0; i < 16; i++)
+    {
+      hex_string += span_id[i];
+    }
+    hex_string += "-";
+    for (int i = 0; i < 2; i++)
+    {
+      hex_string += trace_flags[i];
+    }
+    setter(carrier, kTraceParent, hex_string);
+  }
+
+  static void InjectImpl(Setter setter, T &carrier, const SpanContext &span_context)
+  {
+    InjectTraceParent(span_context, carrier, setter);
+  }
+
+  static bool IsValidHex(nostd::string_view string_view)
+  {
+    for (int i = 0; i < string_view.length(); i++)
+    {
+      if (!(string_view[i] >= '0' && string_view[i] <= '9') &&
+          !(string_view[i] >= 'a' && string_view[i] <= 'f'))
+        return false;
+    }
+    return true;
+  }
+
+  static SpanContext ExtractContextFromTraceParent(nostd::string_view trace_parent)
+  {
+    if (trace_parent.length() != kHeaderSize || trace_parent[kVersionBytes] != '-' ||
+        trace_parent[kVersionBytes + kTraceIdBytes + 1] != '-' ||
+        trace_parent[kVersionBytes + kTraceIdBytes + kSpanIdBytes + 2] != '-')
+    {
+      std::cout << "Unparseable trace_parent header. Returning INVALID span context." << std::endl;
+      return SpanContext(false, false);
+    }
+    nostd::string_view version = trace_parent.substr(0, kHeaderElementLengths[0]);
+    nostd::string_view trace_id =
+        trace_parent.substr(kHeaderElementLengths[0] + 1, kHeaderElementLengths[1]);
+    nostd::string_view span_id = trace_parent.substr(
+        kHeaderElementLengths[0] + kHeaderElementLengths[1] + 2, kHeaderElementLengths[2]);
+    nostd::string_view trace_flags = trace_parent.substr(
+        kHeaderElementLengths[0] + kHeaderElementLengths[1] + kHeaderElementLengths[2] + 3);
+
+    if (version == "ff" &&
```

Comment:
Before you had `if (version == "ff")` and `if (trace_id == "0...0" || span_id == "0...0")` both returning `SpanContext(false, false)`.

Now you've changed this to `if (version == "ff" && (trace_id == "0...0" || span_id == "0...0"))`.

Reply:
Also your test coverage is very minimal. I would expect this change to be caught by a test. Please add more tests.

Reply:
I am trying to add more test and am already fixed the version problem